DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -560,6 +564,83 @@ bool muon::isGoodMuon( const reco::Muon& muon,
     if ( nMatch >= minNumberOfMatches ) return true;
     else return false;
   } // RPCMu
+    
+  if ( type == ME0Mu )
+  {
+    if ( minNumberOfMatches == 0 ) return true;
+    
+    int nMatch = 0;
+    for ( const auto& chamberMatch : muon.matches() )
+    {
+      if ( chamberMatch.detector() != 5 ) continue;
+
+      const double trkX = chamberMatch.x;
+      const double errX = chamberMatch.xErr;
+      const double trkY = chamberMatch.y;
+      const double errY = chamberMatch.yErr;
+
+      for ( const auto& segment : chamberMatch.me0Matches )
+      {
+          
+        const double me0X = segment.x;
+        const double me0ErrX = segment.xErr;
+        const double me0Y = segment.y;
+        const double me0ErrY = segment.yErr;
+
+        const double dX   = fabs(me0X - trkX);
+        const double dY   = fabs(me0Y - trkY);
+        const double pullX   = dX/std::sqrt(errX + me0ErrX);
+        const double pullY   = dY/std::sqrt(errY + me0ErrY);
+          
+        if ( (dX < maxAbsDx or pullX < maxAbsPullX) and (dY < maxAbsDy or pullY < maxAbsPullY) )
+        {
+          ++nMatch;
+          break;
+        }
+      }
+    }
+
+    if ( nMatch >= minNumberOfMatches ) return true;
+    else return false;
+  } // ME0Mu
+  if ( type == GEMMu )
+  {
+    if ( minNumberOfMatches == 0 ) return true;
+    
+    int nMatch = 0;
+    for ( const auto& chamberMatch : muon.matches() )
+    {
+      if ( chamberMatch.detector() != 4 ) continue;
+
+      const double trkX = chamberMatch.x;
+      const double errX = chamberMatch.xErr;
+      const double trkY = chamberMatch.y;
+      const double errY = chamberMatch.yErr;
+
+      for ( const auto& segment : chamberMatch.gemMatches )
+      {
+          
+        const double gemX = segment.x;
+        const double gemErrX = segment.xErr;
+        const double gemY = segment.y;
+        const double gemErrY = segment.yErr;
+
+        const double dX   = fabs(gemX - trkX);
+        const double dY   = fabs(gemY - trkY);
+        const double pullX   = dX/std::sqrt(errX + gemErrX);
+        const double pullY   = dY/std::sqrt(errY + gemErrY);
+          
+        if ( (dX < maxAbsDx or pullX < maxAbsPullX) and (dY < maxAbsDy or pullY < maxAbsPullY) )
```

Comment:
well for the moment this is a very generic function, I implemented the possibility to cut also on local y but it is not said one has to do it. in the past anyway this was considered as an option, not the best one because PullY performed better